src/test_fixture/world.rs

```diff
@@ -150,7 +150,7 @@ impl TestWorld {
     }
 
     #[must_use]
-    pub fn execution_step(execution: usize) -> impl Substep {
+    pub fn execution_step(execution: usize) -> impl Step {
```

Comment:
I just realized that this is going to hurt quite a bit when we stop using `GenericStep`.  I think that we're OK with that; we can just not use this feature for non-generic steps, but @akoshelev might not realize his dream of running tests with non-generic steps as a result.  We rely on this to make steps diverse.

Reply:
Looked into it as well - it seems we can make it work if we consume the world to run a test (that shouldn't require narrowing)

```rust
pub trait Runner {
    async fn oneshot<I, A, O, H, R>(self, input: I, helper_fn: H) -> [O; 3];
    ...
}
```

agree with @martinthomson that we don't need this feature for all tests, likely we only need to run IPA with non generic step